plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/ServerPortExtensionTest.java

```diff
@@ -0,0 +1,284 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.localserver.ui;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.beans.PropertyChangeEvent;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.wst.server.core.IServerType;
+import org.eclipse.wst.server.core.IServerWorkingCopy;
+import org.eclipse.wst.server.ui.wizard.ServerCreationWizardPageExtension.UI_POSITION;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ServerPortExtensionTest {
+
+  private Shell shell;
+  private ServerPortExtension portExtension;
+
+  @Mock private IServerWorkingCopy server;
+
+  @Before
+  public void setUp() {
+    // TODO(chanseok): use ShellTestResource (see AccountPanelTest) after fixing
+    // https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/771.
+    // (Remove shell.dispose() in tearDown() too.)
+    shell = new Shell(Display.getDefault());
+    shell.setVisible(true);
+    portExtension = new ServerPortExtension();
+    portExtension.createControl(UI_POSITION.BOTTOM, shell);
+    portExtension.setServerWorkingCopy(server);
+  }
+
+  @After
+  public void tearDown() {
+    shell.dispose();
+  }
+
+  @Test
+  public void testCreateControl() {
+    assertNotNull(portExtension.portLabel);
+    assertNotNull(portExtension.portText);
+    assertFalse(portExtension.portLabel.isVisible());
+    assertFalse(portExtension.portText.isVisible());
+    assertFalse(portExtension.portDecoration.isVisible());
+  }
+
+  @Test
+  public void testDefaultPort() {
+    assertEquals("8080", portExtension.portText.getText());
+  }
+
+  @Test
+  public void testPortText_allowEmpty() {
+    portExtension.portText.setText("");
+    assertTrue(portExtension.portText.getText().isEmpty());
+  }
+
+  @Test
+  public void testPortText_preventNonNumericCharacters() {
+    portExtension.portText.setText("80A80");
+    assertEquals("8080", portExtension.portText.getText());
+
+    portExtension.portText.setText("80 clipboard pasted 80");
+    assertEquals("8080", portExtension.portText.getText());
+  }
+
+  @Test
+  public void testPortText_numericCharacters() {
+    portExtension.portText.setText("808");
```

Comment:
I can see little value in testing for these two specific values, not sure what this test demonstrates. To test that numerical value is accepted, one of them is enough. How long would a test run that would loop over 0..65535 so the test can verify that all possible port numbers are valid?